docroot/themes/humsci/su_humsci_theme/js/hover_menu.js

```diff
@@ -29,25 +29,67 @@
         }
       }
 
+      // Check main menu items to ensure dropdown submenus don't get orphaned or cut off.
+      function menuEdgeCheck() {
```

Comment:
displace.js may be helpful for what you are trying to achieve here. https://www.drupal.org/node/1956804

Reply:
Decided after discussion against using displace.js, as it would not bring anything to the table that is not resolved by the script I wrote.